### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by learnwithalfred**
## The following requirements have been completed

- Books API Data loaded using fetch ✔️ 
- Fetch book list from Bookstore API on Books page load.✔️ 
- Add an Action Creator, that returns a function.✔️ 
- Dispatch another action (add an action type) that will pass the API response to the reducer and update the state. ✔️ 
- Refactor your add book and remove book features to persist your changes in the server.✔️ 
- Change your existing Action Creators, so they return functions.✔️ 
- Dispatch an action (action type that existed before refactor) that will update the state accordingly. ✔️ 
- Add the book and remove the book should work in the same way after the refactor. ✔️ 

https://user-images.githubusercontent.com/34389326/178613037-d50607b2-bb83-45bf-9251-ce7e52ad4506.mov

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,42 +1,119 @@
...
+
+export const deleteBook = (id) => (dispatch) => {
+  fetch(`${URL}/${id}`, {
+    method: 'DELETE',
+    body: JSON.stringify({ item_id: id }),
+  }).then(() => {
+    dispatch(removeBook(id));
+  });
+};
+
```
Comment (andersonlebon) [inline src/redux/books/books.js]: - [ ] Kindly as the requirements says make sure you use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls.

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @learnwithalfred 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (learnwithalfred) [reply src/redux/books/books.js]: You lied to me

it took more than five hours to fix it
But you said am almost done

Comment (dasileker) [reply src/redux/books/books.js]: 🤣

**Review by dasileker — APPROVED** [review]
### **Hi @learnwithalfred,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by learnwithalfred** [comment]
Thank you so much for the review

**Pull request merged**